File: application/frontend/trpo8-web-client/src/main/webapp/resources/js/app/service/auth/authService.js
/**
 * Authentication service module
 * 
 * @author imoiseyenko@gmail.com
 */
define(["angular", "app/repository/auth/authRepository"], function (angular, authRepositoryModule) {

	var authServiceModule = angular.module("authServiceModule", ["authRepositoryModule"]);

	authServiceModule.factory("authService", ["authRepository", "lsService", function (authRepository, lsService) {

		return {

			// check if user have access token
			isLoggedIn: function () {

				return lsService.getToken() ? true : false;
			},

			signUp: function (email, password, successHandler, errorHandler) {

				return authRepository.signUp(email, password, successHandler, errorHandler);
			},

			signIn: function (email, password, successHandler, errorHandler) {

				return authRepository.signIn(email, password, successHandler, errorHandler);
			},

			// Sign out user from application
			signOut: function (eventsTokenData, successHandler, errorHandler) {

				return authRepository.signOut(eventsTokenData, successHandler, errorHandler);
			}
		};
	}]);

	return authServiceModule;
});

File: application/backend/service/profile-service/spastructure-profile-service-api/src/main/java/com/imoiseyenko/spastructure/backend/service/profile/AuthService.java
package com.imoiseyenko.spastructure.backend.service.profile;

import com.imoiseyenko.spastructure.backend.service.profile.exception.SpastructureProfileServiceException;

/**
 * Auth service interface.
 * 
 * @author imoiseyenko93@gmail.com
 * @version Dec 18, 2014
 */
public interface AuthService {

	/**
	 * Sign out.
	 * 
	 * @param accessToken
	 * @param email
	 * @throws SpastructureProfileServiceException
	 * @author imoiseyenko93@gmail.com
	 */
	public void signOut (String accessToken, String email) throws SpastructureProfileServiceException;
}
